src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java

```diff
@@ -271,15 +271,16 @@ private String getUnqualifiedTypeName() {
   @VisibleForTesting
   static String getParentPattern(String pattern) {
     List<String> segments = getSegments(pattern);
-    int index = segments.size() - 2;
-    while (index >= 0 && !isVariableBinding(segments.get(index))) {
+    int index = segments.size() - 1;
```

Comment:
Is this method ever called for a pattern of "*"? If so, it should return "*" at that point.

Reply:
That's a good point!